index.rst

```diff
@@ -58,24 +58,216 @@ We describe the design and implementation of the LSST Alert Distribution System,
+	"name": "alert",
+	"doc": "sample avro alert schema v1.0",
+	"fields": [
+		{"name": "alertId", "type": "long", "doc": "add descriptions like this"},
+		{"name": "l1dbId", "type": "long"},
+		{"name": "diaSource", "type": "lsst.alert.diaSource"},
+		{"name": "prv_diaSources", "type": [{
+				"type": "array",
+				"items": "lsst.alert.diaSource"}, "null" ], "default": null},
+		{"name": "diaObject", "type": ["lsst.diaObject", "null"], "default": null},
+		{"name": "ssObject", "type": ["lsst.ssObject", "null"], "default": null},
+		{"name": "diaObjectL2", "type": ["lsst.diaObject", "null"], "default": null},
+		{"name": "diaSourcesL2", "type": [{
+				"type": "array",
+				"items": "lsst.alert.diaSource"}, "null"], "default": null},
+		{"name": "cutoutDifference", "type": ["lsst.alert.cutout", "null"], "default": null},
+		{"name": "cutoutTemplate", "type": ["lsst.alert.cutout", "null"], "default": null}
+			]
+    }
+
+
+Full example schemas for LSST alerts can be found in the GitHub repository at
+https://github.com/lsst-dm/sample-avro-alert.
+These were generated using the fields in the ``cat`` package and defined
+by the Data Products Definition Document (LSE-163, DPDD).
+Avro files with realistic data generated by the Simulations team are
+available in data directory of the GitHub repository at
+https://github.com/lsst-dm/alert_stream.
+
 
 
 Alert Distribution
 ==================
 
+Alert distribution uses Apache Kafka, an open source streaming platform
+that can be used for real-time and continuous data pipelines.
+Kafka is a scalable pub/sub message queue based on a commit log.
+It is used in production at scale at companies such as LinkedIn,
+Netflix, and Microsoft to process over 1 trillion messages per day.
+
+Kafka collects messages from processes called "producers,"
+which are organized into distinct streams called "topics."
+Downstream "consumers" pull messages by subscribing to topics.
+Topics can be split into "partitions" that may be distributed
+across multiple machines and allow consumers to read in
+parallel as "consumer groups."
+Data can be replicated by deploying Kafka in cluster mode over several
+servers called "brokers."
+We will refer to these brokers below as "Kafka brokers" to distinguish
+from the LSST alert downstream "community brokers."
+
+For LSST alert distribution, Kafka and the accompanying Zookeeper
+can be deployed as Docker containers from the DockerHub image repository
+maintained by Confluent Inc., the team that created Kafka.
+The latest release of ``alert_stream`` uses Kafka and Zookeeper from
+Confluent platform release 4.1.1.
+The producer used for generating and sending data to Kafka and
+template scripts for consumers of the stream are provided in the GitHub
+repository at https://github.com/lsst-dm/alert_stream,
+which can also be built as a Docker image and deployed as containers.
 
 Alert Filtering
 ================
 
+Alert filters can be written using simple Python functions that act
```

Comment:
Again I'd expand on this: what are the advantages of having users able to write Python filters?  And explain the corresponding challenges in managing user-provided processes.  Hence motivating the choice to run single filters per container.

Reply:
Ok, wrote some text here in the alert filtering section.